recipes/vorbis/all/conanfile.py

```diff
@@ -10,66 +9,80 @@ class VorbisConan(ConanFile):
-        if self.settings.compiler == "Visual Studio":
-            if self.options.shared:
-                self.copy(pattern="*.dll", dst="bin", keep_path=False)
-            self.copy(pattern="*.lib", dst="lib", keep_path=False)
-        else:
-            if self.options.shared:
-                if self.settings.os == "Macos":
-                    self.copy(pattern="*.dylib", dst="lib", keep_path=False)
-                elif self.settings.os == "Windows":
-                    self.copy(pattern="*.dll.a", dst="lib", keep_path=False)
-                    self.copy(pattern="*.dll", dst="bin", keep_path=False)
-                else:
-                    self.copy(pattern="*.so*", dst="lib", keep_path=False)
-            else:
-                self.copy(pattern="*.a", dst="lib", keep_path=False)
-        for pdb_file in glob.glob(os.path.join(self.package_folder, "bin", "*.pdb")):
-            os.unlink(pdb_file)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
 
     def package_info(self):
-        self.cpp_info.libs = ['vorbisfile', 'vorbisenc', 'vorbis']
+        self.cpp_info.names["cmake_find_package"] = "Vorbis"
+        self.cpp_info.names["cmake_find_package_multi"] = "Vorbis"
+        # vorbis
+        self.cpp_info.components["vorbismain"].names["cmake_find_package"] = "vorbis"
+        self.cpp_info.components["vorbismain"].names["cmake_find_package_multi"] = "vorbis"
+        self.cpp_info.components["vorbismain"].names["pkg_config"] = "vorbis"
+        self.cpp_info.components["vorbismain"].libs = ["vorbis"]
         if self.settings.os == "Linux":
-            self.cpp_info.system_libs.append("m")
-        self.cpp_info.names["cmake_find_package"] = 'VORBIS'
-        self.cpp_info.names["cmake_find_package_multi"] = 'VORBIS'
-        self.cpp_info.names['pkg_config'] = 'vorbis'
+            self.cpp_info.components["vorbismain"].system_libs.append("m")
+        self.cpp_info.components["vorbismain"].requires = ["ogg::ogg"]
+        # vorbisenc
+        self.cpp_info.components["vorbisenc"].names["cmake_find_package"] = "vorbisenc"
+        self.cpp_info.components["vorbisenc"].names["cmake_find_package_multi"] = "vorbisenc"
+        self.cpp_info.components["vorbisenc"].names["pkg_config"] = "vorbisenc"
+        self.cpp_info.components["vorbisenc"].libs = ["vorbisenc"]
+        self.cpp_info.components["vorbisenc"].requires = ["vorbismain"]
+        # vorbisfile
+        self.cpp_info.components["vorbisfile"].names["cmake_find_package"] = "vorbisfile"
+        self.cpp_info.components["vorbisfile"].names["cmake_find_package_multi"] = "vorbisfile"
+        self.cpp_info.components["vorbisfile"].names["pkg_config"] = "vorbisfile"
+        self.cpp_info.components["vorbisfile"].libs = ["vorbisfile"]
+        self.cpp_info.components["vorbisfile"].requires = ["vorbismain"]
```

Comment:
```suggestion
        self.cpp_info.components["vorbisfile"].requires = ["vorbismain"]

        # VorbisConfig.cmake defines components 'Enc' and 'File',
        # which create the imported targets Vorbs::vorbisenc and Vorbis::vorbisfile
        self.cpp_info.components["Enc"].requires.append("libvorbisenc")
        self.cpp_info.components["File"].requires.append("libvorbisenc")
```

Reply:
In config file (or target file included by config file), which line creates Enc and File components as alias of vorbisenc and vorbisfile components?

Reply:
See https://github.com/xiph/vorbis/blob/master/cmake/VorbisConfig.cmake.in

Also, I accidentally created a vorbis pr myself before I saw your pr:
https://github.com/conan-io/conan-center-index/pull/2605

Reply:
libsndfile cmake is using these components

Reply:
Also note the 2nd vorbis patch in #2605, to 100% find the ogg dependency.

Reply:
Still don't understand where Vorbis::Enc and Vorbis::File import libs are created (not in VorbisTargets). Some magic in check_required_components?

Reply:
Vorbos::Enc and Vorbis::File are not generated.

`Enc` and `File` are the names of the components that check whether `Vorbis::vorbisenc` and `Vorbis::vorbisfile` exist.
Also, `Vorbis_Enc_FOUND` and `Vorbis_File_FOUND` should be initialized.

Reply:
The name of the component != name of the target.

Reply:
understood ;)

There is a typo, it's not `libvorbisenc` 2 times, but `vorbisenc` and `vorbisfile`. Also I would suggest to give less generic conan component name for Enc and File (possible side effects like name collision for `pkg_config` generator) and explicitly set `cmake_find_package` and `cmake_find_package_multi` names.

Reply:
Cool!
Yeah, using `Enc` as a component name is very generic. Something longer and settings only the cmake names would avoid potential collisions.